langlib/langlib-test/src/test/resources/test-src/arraylib_test.bal

```diff
@@ -1724,3 +1729,79 @@ function testArrSortWithNamedArgs3() {
     });
     assertValueEquality([1, 10, 0, "AAD", "ABC", "ADS", "DES"], sortedArr);
 }
+
+function testPushWithObjectConstructorExpr() {
+    Obj[] arr = [];
+    arr.push(object {
+        int i = 123;
+        int j = 234;
+    });
+    assertValueEquality(1, arr.length());
+
+    Obj ob = arr[0];
+    assertValueEquality(123, ob.i);
+    assertValueEquality(234, ob.j);
+}
+
+type Template object {
+    *object:RawTemplate;
+
+    public (readonly & string[]) strings;
+    public int[] insertions;
+};
```

Comment:
This type is created to become similar to the `sql:ParameterizedQuery` ryt?.
So shall we use distinct keyword also.
because in SQL module it is like.
```
public type ParameterizedQuery distinct object {
```

Reply:
Not really. This fails with any raw template type that overrides one or both of the fields. The current tests already test type checking, so IMO, they should suffice.